ssg/boolean_expression.py

```diff
@@ -85,17 +88,29 @@ def __lt__(self, other):
 
     def as_id(self):
         id_str = self.name
-        for (op, ver) in self.spec.specs:
+        for (op, ver) in self.version_definitions:
             id_str += '_{0}_{1}'.format(SPEC_OP_ID_TRANSLATION.get(op, 'unknown_spec_op'), ver)
+        if self.arg:
+            id_str += '_' + self.arg
         return id_str
 
+    def as_dict(self):
+        res = {'id': self.as_id(), 'name': self.name, 'arg': ''}
+        if self.arg:
+            res['arg'] = self.arg
+        return res
+
     @property
-    def name(self):
-        return self.spec.project_name
+    def arg(self):
+        return self.requirement.extras[0] if self.requirement.extras else None
 
     @property
-    def specs(self):
-        return self.spec.specs
+    def version_definitions(self):
```

Comment:
This property isn't mentioned in the PR description, and not tested either. However, it is used in quite a complex way in the new code.
Looking at other places, I would probably skip this method, and go for a method that converts underlying version specifiers to strings, because this is what this property is used for ATM.

Reply:
It should not be there at all. We are not introducing versions ATM.

Reply:
So I would suggest to remove it in as subsequent PR, or to change the approach - the code used in `as_id` and in `PlatformSymbol.as_cpe_lang_xml` is both similar to each other, and both places depend on a particular representation of the version spec.